basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
+                    if(wm->getMapRoutingGraph()->right(starting_lane[0])){
+                        curr_end_lanelet = wm->getMapRoutingGraph()->right(starting_lane[0]).get();
+                    }
+                    else{
+                        curr_end_lanelet = wm->getMapRoutingGraph()->adjacentRight(starting_lane[0]).get();
+                    }
+                }
+                
+                auto ending_lane_linestring = curr_end_lanelet.centerline2d().basicLineString();
+                ending_lane_centerline.insert(ending_lane_centerline.end(), ending_lane_linestring.begin(), ending_lane_linestring.end());
+                
+            }
+
+            cav_msgs::VehicleState ending_state = waypoint_generation::get_nearest_state_to_downtrack(ending_lane_centerline, ending_downtrack, wm);
+
+            int reference_line_start_index = waypoint_generation::get_nearest_point_index(reference_centerline, state);
+            int ending_line_end_index = waypoint_generation::get_nearest_point_index(ending_lane_centerline, ending_state);
+
+            lanelet::BasicLineString2d starting_centerline(reference_centerline.begin() + reference_line_start_index, reference_centerline.end());
+            lanelet::BasicLineString2d ending_centerline(ending_lane_centerline.begin(), ending_lane_centerline.begin() + ending_line_end_index);
+
+            //Convert to vector of BasicPoint2d
+
+                //Downsample centerlines
+            lanelet::BasicLineString2d downsampled_start_centerline;
+            downsampled_start_centerline.reserve(400);
+            downsampled_start_centerline = carma_utils::containers::downsample_vector(starting_centerline, 5);
+            std::vector<lanelet::BasicPoint2d> start_centerline;
+            for(int i = 0;i<starting_centerline.size();++i){
+                start_centerline.push_back(starting_centerline[i]);
+            }
+
+            lanelet::BasicLineString2d downsampled_end_centerline;
+            downsampled_end_centerline.reserve(400);
+            downsampled_end_centerline = carma_utils::containers::downsample_vector(ending_centerline, 5);
+            std::vector<lanelet::BasicPoint2d> end_centerline;
+            for(int i = 0; i<ending_centerline.size();++i){
+                end_centerline.push_back(ending_centerline[i]);
+            }
+            //If points are not the same size - ensure same size along both centerlines
+            if(start_centerline.size() != end_centerline.size())
+            {
+                //Resample centerlines to same length
+                auto centerlines = resample_linestrings_to_same_length(start_centerline, end_centerline);
+                start_centerline = centerlines.front();
+                end_centerline = centerlines.back();
+
+            }
+
+            //Create Trajectory geometry
+            double delta_step = 1.0 / start_centerline.size();
+
+            for (int i = 0; i < start_centerline.size(); i++)
```

Comment:
For all these positive iterators use size_t instead of int to avoid compiler/sonar cloud warnings